tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/AWS/AwsS3Tests.cs

```diff
@@ -0,0 +1,111 @@
+// <copyright file="AwsS3Tests.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Datadog.Trace.Configuration;
+using Datadog.Trace.TestHelpers;
+using FluentAssertions;
+using VerifyXunit;
+using Xunit;
+using Xunit.Abstractions;
+
+namespace Datadog.Trace.ClrProfiler.IntegrationTests.AWS
+{
+    [Trait("RequiresDockerDependency", "true")]
+    [UsesVerify]
+    public class AwsS3Tests : TracingIntegrationTest
+    {
+        public AwsS3Tests(ITestOutputHelper output)
+            : base("AWS.S3", output)
+        {
+        }
+
+        public static IEnumerable<object[]> GetEnabledConfig()
+            => from packageVersionArray in PackageVersions.AwsS3
+               from metadataSchemaVersion in new[] { "v0", "v1" }
+               select new[] { packageVersionArray[0], metadataSchemaVersion };
+
+        public override Result ValidateIntegrationSpan(MockSpan span, string metadataSchemaVersion) => span.Tags["span.kind"] switch
+        {
+            SpanKinds.Client => span.IsAwsS3Request(metadataSchemaVersion),
+            _ => throw new ArgumentException($"span.Tags[\"span.kind\"] is not a supported value for the AWS S3 integration: {span.Tags["span.kind"]}", nameof(span)),
+        };
+
+        [SkippableTheory]
+        [MemberData(nameof(GetEnabledConfig))]
+        [Trait("Category", "EndToEnd")]
+        public async Task SubmitsTraces(string packageVersion, string metadataSchemaVersion)
+        {
+            SetEnvironmentVariable("DD_TRACE_SPAN_ATTRIBUTE_SCHEMA", metadataSchemaVersion);
+            var isExternalSpan = metadataSchemaVersion == "v0";
+            var clientSpanServiceName = isExternalSpan ? $"{EnvironmentHelper.FullSampleName}-aws-s3" : EnvironmentHelper.FullSampleName;
+
+            using var telemetry = this.ConfigureTelemetry();
+            using (var agent = EnvironmentHelper.GetMockAgent())
+            using (await RunSampleAndWaitForExit(agent, packageVersion: packageVersion))
+            {
+#if NETFRAMEWORK
+                var expectedCount = 8;
+                var frameworkName = "NetFramework";
+#else
+                var expectedCount = 4;
+                var frameworkName = "NetCore";
+#endif
+                var spans = agent.WaitForSpans(expectedCount);
```

Comment:
nitpicky, but two things:

`expectedCount` is incorrect, there are definitely more spans that we are expecting (and getting) could both .NET Core and .NET Framework get updated? I expect these would also change when adding in the additional methods for the missed instrumentations.

Additionally (not as important IMO), could you add a check for `spans.Count().Should().Be(expectedCount)`?